libpimeval/src/pimDevice.cpp

```diff
@@ -228,13 +282,81 @@ pimDevice::init(PimDeviceEnum deviceType, const char* configFileName)
   m_resMgr = new pimResMgr(this);
 
   m_cores.resize(m_numCores, pimCore(m_numRows, m_numCols));
+  std::printf("PIM-Info: Current Device = %s, Simulation Target = %s\n",
+              pimUtils::pimDeviceEnumToStr(m_deviceType).c_str(),
```

Comment:
Where do you assign the m_deviceType variable?

Reply:
It is still unclear for me how to resolve the value of m_deviceType based on m_simTarget. Could you please describe how these two variables are mapped together?

Reply:
Given a simTarget, we need to know if the application is running functional simulation or micro-ops level simulation. In functional simulation, the application can only use high-level functional APIs. In micro-ops level simulation, the application can only use architecture-specific micro-ops. We'll add device type sanity checks to limit the usage later. An application can create a device use PIM_FUNCTIONAL to run functional simulation and a specific device type to run micro-ops level simulation. For now, we can make sure the config file can only affect applications created with PIM_FUNCTIONAL for now.